src/repository.ts

```diff
@@ -0,0 +1,393 @@
+    return validations;
+  }
+
+  public prepare(opts?: PrepareOpts): void {
+    if (opts.dryrun) {
+      const cmd = 'npx lerna version --conventional-commits --yes --no-commit-hooks --no-push --no-git-tag-version';
+      this.execCommand(cmd);
+      this.revertUnstagedChanges();
+    } else {
+      const cmd =
+        'npx lerna version --conventional-commits --yes --no-commit-hooks --no-push --message "chore(release): publish [ci skip]"';
+      this.execCommand(cmd);
+    }
+  }
+
+  public async sign(packageNames: string[]): Promise<SigningResponse[]> {
+    const packages = this.packages.filter((pkg) => packageNames.includes(pkg.name));
+    const responses: SigningResponse[] = [];
+    const signer = await Signer.create(this.ux);
+    for (const pkg of packages) {
+      this.ux.log(chalk.dim(`Signing ${pkg.name} at ${pkg.location}`));
+      const response = await signer.sign(pkg.location);
+      responses.push(response);
+    }
+    return responses;
+  }
+
+  public publish(opts: PublishOpts): void {
+    const { dryrun, signatures, access, tag } = opts;
+    const tarPathsByPkgName: { [key: string]: string } = signatures.reduce((res, curr) => {
+      res[curr.name] = curr.tarPath;
+      return res;
+    }, {});
+    for (const pkg of this.packages) {
+      const tarPath = tarPathsByPkgName[pkg.name];
+      let cmd = 'npm publish';
+      if (tarPath) cmd += ` ${tarPath}`;
+      if (tag) cmd += ` --tag ${tag}`;
+      if (dryrun) cmd += ' --dry-run';
+      cmd += ` --access ${access || 'public'}`;
+      this.execCommand(`(cd ${pkg.location} ; ${cmd})`);
+    }
+  }
+
+  public async waitForAvailability(): Promise<boolean> {
+    let found = false;
+    let attempts = 0;
+    const maxAttempts = 300;
+    this.ux.startSpinner('Polling for new versions to become available on npm');
+    while (attempts < maxAttempts && !found) {
+      attempts += 1;
+      this.ux.setSpinnerStatus(`attempt: ${attempts} of ${maxAttempts}`);
+      found = this.packages.every((pkg) => pkg.nextVersionIsAvailable());
+      await wait(1000);
+    }
+    this.ux.stopSpinner(attempts >= maxAttempts ? 'failed' : 'done');
+    return found;
+  }
+
+  public verifySignature(packageNames: string[]): void {
+    const packages = this.packages.filter((pkg) => packageNames.includes(pkg.name));
+    for (const pkg of packages) {
+      const cmd = `sfdx-trust plugins:trust:verify --npm ${pkg.name}@${pkg.getNextVersion()}`;
+      this.execCommand(cmd);
+    }
+  }
+
+  public getSuccessMessage(): string {
+    const successes = this.packages.map((pkg) => `  - ${pkg.name}@${pkg.getNextVersion()}`).join(os.EOL);
+    const header = chalk.green.bold(`${os.EOL}Successfully published:`);
+    return `${header}${os.EOL}${successes}`;
+  }
+
+  protected async init(): Promise<void> {
+    this.logger = await Logger.child(this.constructor.name);
+    const pkgPaths = await getListOfPackages();
+    const nextVersions = this.determineNextVersionByPackage();
```

Comment:
Hmm, lerna supports publishing only packages that have changed. This looks like it will always try to publish all packages regardless if the version has changed?